M0090IDE-4172 Document the new test capabilities of the SDK 6 (#583)
MicroEJ/docs: SDK6UserGuide/testProject.rst

The SDK provides the capabilities to implement and run tests for a project.
It relies on the standard `JUnit`_ API.

There are different types of tests:

- test on the Simulator
- test on a device
- test on a J2SE VM

Each type of test is detailed in the next sections.

.. _JUnit: https://repository.microej.com/modules/ej/library/test/junit/

JUnit Compliance
----------------

The SDK relies on `JUnit <https://junit.org/junit4/>`__, the most popular Java testing framework, to define and execute the tests.
It supports a subset of JUnit 4, namely the annotations: ``@After``,
``@AfterClass``, ``@Before``, ``@BeforeClass``, ``@Ignore``, ``@Test``.

Each test case entry point must be declared using the ``org.junit.Test`` annotation (``@Test`` before a method declaration). 

> Refer to `JUnit documentation <https://junit.org/junit4/>`__ to get details on theusage of other annotations.

Reviewer: ```suggestion
Refer to `JUnit documentation <https://junit.org/junit4/>`__ to get details on the usage of other annotations.
```

Author: Done.

- ``(1)``: configures all the testsuites of the project.
- ``(2)``: configures the built-in ``test`` suite provided by Gradle. Use this testsuite to configure the tests on the Simulator.
- ``(3)``: declares that this testsuite uses the MicroEJ Test Engine. By default, the MicroEJ Test Engine executes the tests on the
  Simulator.
- ``(4)``: adds the dependencies required by the tests. The first line declares a dependency to the code of the project.
  The second line declares a dependency on the ``edc`` Library. The third line declares a dependency to the JUnit API used 
  to annotate Java Test classes. 

Create a Test Class
~~~~~~~~~~~~~~~~~~~

The SDK provides a JUnit library containing the subset of the supported JUnit API: ``ej.library.test:junit``.
Before creating the Test class, make sure this library is declared in the testsuite dependencies:

.. code-block::

   testing {
      suites {
         val test by getting(JvmTestSuite::class) {
            ...
            dependencies {
               implementation("ej.library.test:junit:1.7.1")
            }
            ...
         }
      }
   }

> The test class can npw be created in the ``src/test/java`` folder.

Reviewer: ```suggestion
The test class can now be created in the ``src/test/java`` folder.
```

Author: Done.

      - right-click on the ``src/test/java`` folder.

>       - select :guilabel:`New` > :guilabel:`Java Class``, then press ``Alt`` + ``Insert`` and select ``Test Method``.

Reviewer: ```suggestion
      - select :guilabel:`New` > :guilabel:`Java Class`, then press ``Alt`` + ``Insert`` and select ``Test Method``.
```

Author: Done.
